tests/e2e/framework/kubernetes.go

```diff
@@ -198,6 +207,15 @@ func newKubeInfo(tmpDir, runID, baseVersion string) (*KubeInfo, error) {
 
 	a := NewAppManager(tmpDir, *namespace, i, kubeConfig)
 
+	clusters := make(map[string]string)
+	appPods := make(map[string]*appPodsInfo)
+	clusters[PrimaryCluster] = kubeConfig
```

Comment:
Since you needed to add the clusters map you shouldn't really need to have both the primary clusters and the remote clusters kubeConfig in two places off the same structure.   I guess make that change though would entail change many places and you might not want to do that at this time.

Reply:
Yeah, we can clean that up in a separate PR. The reason for me to add 'clusters' field is that it really makes it easy to change the test code.